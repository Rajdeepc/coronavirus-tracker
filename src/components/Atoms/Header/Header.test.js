import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from './Header.js'; // importing the file to test

afterEach(cleanup);

it('renders', () => {
    const { asFragment } = render(<Header text= "I am header" />) // asFramgment return html for snapshot testing
    expect(asFragment()).toMatchSnapshot();
});

it('inserts text in h1', () => {
    const { getByTestId, getByText } = render(<Header text="I am header" />);
    expect(getByTestId('h1tag')).toHaveTextContent('I am header');
    expect(getByText('I am header')).toHaveClass('fancy-h1');
})