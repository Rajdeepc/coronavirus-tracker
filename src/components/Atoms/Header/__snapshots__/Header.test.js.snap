// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="fancy-h1"
      data-testid="h1tag"
    >
      I am header
    </h1>
  </header>
</DocumentFragment>
`;
